### ISSUE TITLE
boto.s3.key.Key.set_contents_from_stream() not documented as unsupported

### ISSUE THREAD

**Issue by collinstocks** [boto/boto#3282]
S3 does not support chunked file transfer. This means that there are *no circumstances* under which boto.s3.key.Key.set_contents_from_stream() will work.

This fact should show up in the documentation.

**Comment by collinstocks** [comment]
I mean, seriously. I just wasted about an hour "improving" my application so that it could take advantage of set_contents_from_stream(), only to find out that this method, which is documented, can never possibly work.

**Comment by collinstocks** [comment]
I'll be happy to change the documentation and send in a pull request if someone wants to come up with some wording.

**Comment by jamesls** [comment]
Yeah, I believe this was added for gs which supports chunked transfer.  Not opposed to updating the documentation to make it clear that Amazon S3 does not support this.
